src/application-manager.coffee

```diff
@@ -0,0 +1,1047 @@
+					.then =>
+						delete @_containerStarted[step.current.containerId]
+						if step.options?.removeImage
+							@images.removeByDockerId(step.current.image)
+			updateMetadata: (step, { force = false, skipLock = false } = {}) =>
+				@_lockingIfNecessary step.current.appId, { force, skipLock: skipLock or step.options?.skipLock }, =>
+					@services.updateMetadata(step.current, step.target)
+			restart: (step, { force = false, skipLock = false } = {}) =>
+				@_lockingIfNecessary step.current.appId, { force, skipLock: skipLock or step.options?.skipLock }, =>
+					Promise.try =>
+						@services.kill(step.current, { wait: true })
+					.then =>
+						delete @_containerStarted[step.current.containerId]
+					.then =>
+						@services.start(step.target)
+					.then (container) =>
+						@_containerStarted[container.id] = true
+			stopAll: (step, { force = false, skipLock = false } = {}) =>
+				@stopAll({ force, skipLock })
+			start: (step) =>
+				@services.start(step.target)
+				.then (container) =>
+					@_containerStarted[container.id] = true
+			handover: (step, { force = false, skipLock = false } = {}) =>
+				@_lockingIfNecessary step.current.appId, { force, skipLock: skipLock or step.options?.skipLock }, =>
+					@services.handover(step.current, step.target)
+			fetch: (step) =>
+				startTime = process.hrtime()
+				@fetchesInProgress += 1
+				Promise.join(
+					@config.get('fetchOptions')
+					@images.getAvailable()
+					(opts, availableImages) =>
+						opts.deltaSource = @bestDeltaSource(step.image, availableImages)
+						@images.triggerFetch step.image, opts, (success) =>
+							@fetchesInProgress -= 1
+							@timeSpentFetching += process.hrtime(startTime)[0]
+							if success
+								@reportCurrentState(update_downloaded: true)
+				)
+			removeImage: (step) =>
+				@images.remove(step.image)
+			saveImage: (step) =>
+				@images.save(step.image)
+			cleanup: (step) =>
+				@images.cleanup()
+			createNetworkOrVolume: (step) =>
+				model = if step.model is 'volume' then @volumes else @networks
+				model.create(step.target)
+			removeNetworkOrVolume: (step) =>
+				model = if step.model is 'volume' then @volumes else @networks
+				model.remove(step.current)
+			ensureSupervisorNetwork: =>
+				@networks.ensureSupervisorNetwork()
+		}
+		@validActions = _.keys(@actionExecutors).concat(@proxyvisor.validActions)
+		@_router = new ApplicationManagerRouter(this)
+		@router = @_router.router
+
+		@images.on('change', @reportCurrentState)
+		@services.on('change', @reportCurrentState)
+
+	serviceAction: serviceAction
+	imageForService: imageForService
+	fetchAction: fetchAction
+
+	reportCurrentState: (data) =>
+		@emit('change', data)
+
+	init: =>
+		@images.cleanupDatabase()
+		.then =>
+			@services.attachToRunning()
+		.then =>
+			@services.listenToEvents()
+
+	# Returns the status of applications and their services
+	# TODO: discuss: I think commit could be deduced by the UI looking at the image_installs on the API?
```

Comment:
I don't think we should worry about this just now, it will soon be a release ID anyway.